packages/flutter/lib/src/material/carousel.dart

```diff
@@ -58,11 +88,42 @@ class CarouselView extends StatefulWidget {
     this.onTap,
     required this.itemExtent,
     required this.children,
-  });
+  }) : allowFullyExpand = true,
+       layoutWeights = null;
+
+/// Creates a scrollable list where the size of each child widget is dynamically
+/// determined by the provided [layoutWeights].
+///
+/// The `layoutWeights` parameter is required and defines the relative size
+/// proportions of each child widget.
+///
+/// When [allowFullyExpand] is set to `true`, each child can be expanded to its
+/// maximum size while scrolling. For example, with [layoutWeights] of `[1, 7, 1]`,
+/// the initial weight of the first item is 1. However, by enabling
+/// [allowFullyExpand] and scrolling forward, the first item can expand to occupy
+/// a weight of 7, leaving a weight of 1 as white space before it. This feature
+/// is particularly useful for achieving "hero" and "center-aligned hero" layouts.
+  const CarouselView.weighted({
+    super.key,
+    this.padding,
+    this.backgroundColor,
+    this.elevation,
+    this.shape,
+    this.overlayColor,
+    this.itemSnapping = false,
+    this.shrinkExtent = 0.0,
+    this.controller,
+    this.scrollDirection = Axis.horizontal,
+    this.reverse = false,
+    this.allowFullyExpand = true,
+    this.onTap,
+    required this.layoutWeights,
```

Comment:
Is it intentional that you could pass null in here or should null be disallowed with:

```suggestion
    required List<int> this.layoutWeights,
```